WordPress/Classes/ViewRelated/NUX/Signin2FAViewController.swift

```diff
@@ -0,0 +1,249 @@
+        let options = [
+            NSDocumentTypeDocumentAttribute: NSHTMLTextDocumentType
+        ]
+
+        let styledString = "<style>body {font-family: sans-serif; font-size:14px; color: #ffffff; text-align:center;}</style>" + string
+        let attributedCode = try! NSMutableAttributedString(data: styledString.dataUsingEncoding(NSUTF8StringEncoding)!,
+                                                            options: options,
+                                                            documentAttributes: nil)
+
+        let attributedCodeHighlighted = attributedCode.mutableCopy() as! NSMutableAttributedString
+        attributedCodeHighlighted.applyForegroundColor(WPNUXUtility.confirmationLabelColor())
+
+        sendCodeButton.titleLabel!.lineBreakMode = .ByWordWrapping
+        sendCodeButton.titleLabel!.textAlignment = .Center
+        sendCodeButton.titleLabel!.numberOfLines = 3
+        sendCodeButton.setAttributedTitle(attributedCode, forState: .Normal)
+        sendCodeButton.setAttributedTitle(attributedCodeHighlighted, forState: .Highlighted)
+    }
+
+
+    /// Displays the specified text in the status label.
+    ///
+    /// - Parameters:
+    ///     - message: The text to display in the label.
+    ///
+    func configureStatusMessage(message: String) {
+        statusLabel.text = message
+
+        sendCodeButton.hidden = !message.isEmpty
+    }
+
+
+    /// Configures the appearance and state of the submit button.
+    ///
+    func configureSubmitButton(animating: Bool) {
+        submitButton.showActivityIndicator(animating)
+
+        submitButton.enabled = (
+            !animating &&
+            !loginFields.multifactorCode.isEmpty
+        )
+    }
+
+
+    /// Sets the view's state to loading or not loading.
+    ///
+    /// - Parameters:
+    ///     - loading: True if the form should be configured to a "loading" state.
+    ///
+    func configureLoading(loading: Bool) {
+        verificationCodeField.enablesReturnKeyAutomatically = !loading
+
+        configureSubmitButton(loading)
+    }
+
+
+    /// Configure the view for an editing state. Should only be called from viewWillAppear
+    /// as this method skips animating any change in height.
+    ///
+    func configureViewForEditingIfNeeded() {
+        // Check the helper to determine whether an editiing state should be assumed.
+        adjustViewForKeyboard(SigninEditingState.signinEditingStateActive)
+        if SigninEditingState.signinEditingStateActive {
+            verificationCodeField.becomeFirstResponder()
+        }
+    }
+    
+
+    // MARK: - Instance Methods
+
+
+    /// Validates what is entered in the various form fields and, if valid,
+    /// proceeds with the submit action.
+    ///
+    func validateForm() {
+        view.endEditing(true)
+
+        // is reachable?
+        if !ReachabilityUtils.isInternetReachable() {
+            ReachabilityUtils.showAlertNoInternetConnection()
+            return
+        }
```

Comment:
I recently learned from [a WWDC talk](https://developer.apple.com/videos/play/wwdc2012/706/) that Apple doesn't recommend using Reachability to *prevent* any connection. You should attempt the connection an if it fails, then use Reachability to determine when to retry.